src/Evolution/Systems/Ccz4/FiniteDifference/SecondPartialDerivatives.cpp

```diff
@@ -0,0 +1,1139 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/Ccz4/FiniteDifference/SecondPartialDerivatives.hpp"
+
+#include <array>
+#include <cstddef>
+#include <limits>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/Transpose.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Structure/DirectionMap.hpp"
+#include "NumericalAlgorithms/FiniteDifference/PartialDerivatives.hpp"
+#include "NumericalAlgorithms/Spectral/Basis.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Quadrature.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/ErrorHandling/Error.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+
+namespace Ccz4::fd {
+namespace {
+template <size_t Order, bool UnitStride>
+struct ComputeImpl;
+
+// only the 4-th order is implemented for now
+template <bool UnitStride>
+struct ComputeImpl<4, UnitStride> {
+  static constexpr size_t fd_order = 4;
+
+  // The stencil is used where ghost data from two neighbors is needed.
+  // It uses 13 points including the point of derivative (the 0th
+  // point). We use 1-4 to label the four points distant \delta from the 0th
+  // point, 5-6 to label the two diagonal points distant sqrt(2)*\delta from the
+  // 0th point, 7-10 to label the four points distant 2*\delta from the 0th
+  // point, and 11-12 to label the two diagonal points distant 2*sqrt(2)*\delta
+  // from the 0th point. Since the weights associated with the points of the
+  // same distance to the 0th point are the same, the ordering with each
+  // subgroup above is arbitrary.
+  SPECTRE_ALWAYS_INLINE static double mixed_second_deriv_special_pointwise(
+      const std::array<double, 13>& q, const int stride,
+      const std::array<double, 3>& weights) {
+    ASSERT(stride == 1, "Only UnitStride is supported" << stride);
+    return weights[0] * q[0] +
+           weights[1] * (q[1] + q[2] + q[3] + q[4] - q[5] - q[6]) +
+           weights[2] * (q[11] + q[12] - q[7] - q[8] - q[9] - q[10]);
+  }
+
+  static constexpr std::array<double, 3>
+  mixed_second_derivative_special_weights(const double one_over_delta_squared) {
+    return {{-1.25 * one_over_delta_squared,
+             0.66666666666666667 * one_over_delta_squared,
+             0.04166666666666667 * one_over_delta_squared}};
```

Comment:
Could you document the non-obvious fractions? e.g. 1/24, etc.? Same for other numbers below

Reply:
Sorry, you're right. All is good!